controllers/util/mw_util.go

```diff
@@ -194,111 +194,71 @@ func (mwu *MWUtil) CreateOrUpdateNamespaceManifest(
 	return mwu.createOrUpdateManifestWork(manifestWork, managedClusterNamespace)
 }
 
-func (mwu *MWUtil) generateNamespaceManifest(name string) (*ocmworkv1.Manifest, error) {
-	return mwu.GenerateManifest(&corev1.Namespace{
+func namespace(name string) *corev1.Namespace {
+	return &corev1.Namespace{
 		TypeMeta:   metav1.TypeMeta{Kind: "Namespace", APIVersion: "v1"},
 		ObjectMeta: metav1.ObjectMeta{Name: name},
-	})
-}
-
-func ClusterRolesList(ctx context.Context, client client.Client, clusterNames *sets.String) error {
-	manifestworks := &ocmworkv1.ManifestWorkList{}
-	if err := client.List(ctx, manifestworks); err != nil {
-		return fmt.Errorf("manifestworks list: %w", err)
-	}
-
-	for i := range manifestworks.Items {
-		manifestwork := &manifestworks.Items[i]
-		if manifestwork.ObjectMeta.Name == ClusterRolesManifestWorkName {
-			*clusterNames = clusterNames.Insert(manifestwork.ObjectMeta.Namespace)
-		}
 	}
-
-	return nil
 }
 
-var clusterRolesMutex sync.Mutex
-
-func (mwu *MWUtil) ClusterRolesCreate(drpolicy *rmn.DRPolicy) error {
-	clusterRolesMutex.Lock()
-	defer clusterRolesMutex.Unlock()
+func (mwu *MWUtil) CreateOrUpdateDrClusterManifestWork(
+	clusterName string,
+	ramenConfig *rmn.RamenConfig,
+	drClusterOperatorChannelName string,
+	drClusterOperatorPackageName string,
+	drClusterOperatorNamespaceName string,
+	drClusterOperatorCatalogSourceName string,
+	drClusterOperatorCatalogSourceNamespaceName string,
+	drClusterOperatorClusterServiceVersionName string,
+) error {
+	objects := []interface{}{
+		vrgClusterRole,
+		vrgClusterRoleBinding,
+	}
+	if ramenConfig.DrClusterOperator.DeploymentAutomationEnable {
+		objects = append(objects,
+			namespace(drClusterOperatorNamespaceName),
+			olmClusterRole,
+			olmRoleBinding(drClusterOperatorNamespaceName),
+			operatorGroup(drClusterOperatorNamespaceName),
+			operatorGroup(drClusterOperatorNamespaceName),
```

Comment:
I just noticed the operator group is listed twice...removing the duplicate now...

Reply:
fixed